protocol/tx/entry.go

```diff
@@ -26,10 +26,17 @@ type entry interface {
 var errInvalidValue = errors.New("invalid value")
 
 func entryID(e entry) (hash bc.Hash) {
+	// This nil test only handles the case where e is the zero value of
+	// the entry interface.
 	if e == nil {
 		return hash
 	}
 
+	// This nil test handles the case where e is a nil pointer.
+	if reflect.ValueOf(e).IsNil() {
```

Comment:
May I suggest combining these into a single test and still including the commentary. Plus maybe a link to the FAQ might obviate some of the verbiage in this file.

```go
// Check for both entryID(nil) and entryID(x)
// where x is a concrete nil pointer.
// See https://golang.org/doc/faq#nil_error.
if e == nil || reflect.ValueOf(e).IsNil() {
	return bc.Hash{}
}
```